csrc/ops/utils.cpp

```diff
@@ -174,12 +174,128 @@ IterType promoteIterType(IterType type1, IterType type2) {
   }
 }
 
+//! For MatmulOp, the input iterdomains at a given index do not necessarily map to the output iterdomain at that index
+//! This function aligns the input iterdomain to the output and returns a vector where each element is the
+//! input iterdomain corresponding to the output iterdomain at that index.
+//! If the element is nullptr, there is no mapping between input-output at that
+//! index. Based on the input dimensions following cases are possible:
+//! 1. A/B is 1D: [M, K] x [K] -> [M]
+//! Mapping A: {id_M}, Mapping B: {nullptr}
+//! 2. A and B are 2D: [M, K] x [K, N] -> [M, N]
+//! Mapping A: {id_M, nullptr}, Mapping B: {nullptr, id_N}
+//! 3. A/B are atleast 1D and one of them is > 2D: [B, M, K] x [K, N] -> [B, M,
+//! N] Mapping A: {id_B, id_M, nullptr}, Mapping B: {nullptr, nullptr, id_N}
+std::vector<IterDomain*> mapMatmulOpIterDomains(
+    const std::vector<IterDomain*>& input_domain,
+    bool is_lhs,
+    size_t out_size) {
+  std::vector<IterDomain*> mapping(out_size, nullptr);
+  auto inp_size = (int64_t)input_domain.size();
+
+  if (inp_size == 1) {
+    // Only reduction axis {K}
+    return mapping;
+  }
+  // Input A to matmul: {*, M, K}
+  // Input B to matmul: {*, K, N} / {K} (ndims = 1)
+  auto kpos = is_lhs ? inp_size - 1 : inp_size - 2;
+
+  // If A/B is 1D, out_size < inp_size.
+  for (auto out_inx = (int64_t)out_size - 1, inp_inx = inp_size - 1; inp_inx >= 0;
```

Comment:
```suggestion
  for (auto out_index = (int64_t)out_size - 1, inp_index = inp_size - 1; inp_index >= 0;
```

Or `idx`. `inx` is a much less common acronym.